flexmeasures/data/models/reporting/__init__.py

```diff
@@ -20,104 +20,25 @@ class Reporter(DataGeneratorMixin):
 
     sensor: Sensor = None
 
-    reporter_config: Optional[dict] = None
-    reporter_config_raw: Optional[dict] = None
-    schema = ReporterConfigSchema()
-    data: Dict[str, Union[tb.BeliefsDataFrame, pd.DataFrame]] = None
-
-    def __init__(
-        self, sensor: Sensor, reporter_config_raw: Optional[dict] = None
-    ) -> None:
-        """
-        Initialize a new Reporter.
-
-        Attributes:
-        :param sensor: sensor where the output of the reporter will be saved to.
-        :param reporter_config_raw: dictionary with the serialized configuration of the reporter.
-        """
-
-        self.sensor = sensor
-
-        if not reporter_config_raw:
-            reporter_config_raw = {}
+    _inputs_schema = ReporterInputsSchema()
+    _config_schema = ReporterConfigSchema()
 
-        self.reporter_config_raw = reporter_config_raw
+    def __init__(self, config: dict | None = None, **kwargs) -> None:
+        super().__init__(config, **kwargs)
 
-    def fetch_data(
-        self,
-        start: datetime,
-        end: datetime,
-        input_resolution: timedelta = None,
-        belief_time: datetime = None,
-    ):
-        """
-        Fetches the time_beliefs from the database
-        """
+        self.sensor = self._config["sensor"]
```

Comment:
The DataGenerator with its config and input paramters allows for a MIMO function description. In the case of the Reporter, generally, the report will only output to a single sensor.

Aside, I need to move this to the inputs instead, so that a change of the output sensor doesn't trigger the creation of a new DataSource.

Reply:
Your last point makes sense to me.

Your first point also makes sense, but just had me thinking. I could imagine we'd would want to create a reporter that outputs a frame with multiple columns, and then each column could need to be mapped to an output sensor (in case the results should be persisted). Or some subset of columns, in case not all columns need to be saved. Having a single reporter create multiple time series (possibly even with some [parallel Pandas operations](https://pypi.org/project/pandarallel)) could be a quite efficient approach. I am already running a lot of reporters that I currently trigger with individual CLI calls, giving quite the overhead in server starts.